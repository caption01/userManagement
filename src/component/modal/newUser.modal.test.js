import React, {Component} from 'react';
import NewUserModal from './newUser.modal.component';
import { shallow } from 'enzyme'

describe('Testing NewUserModal', () => {

    it('expect render newUserModal', () => {

        const wrapper = shallow(<NewUserModal />)
        expect(wrapper).toMatchSnapshot()
    })

    it('testing handleOk function for null input', () => {
        const mockProps = {
            addNewUser: jest.fn(),
            setActiveModal: jest.fn() 
        }
        window.alert = jest.fn();
        const wrapper = shallow(<NewUserModal {...mockProps} />)

        wrapper.find('[id="inputfirstname"]').simulate('change', { target: { value: '' } })
        wrapper.find('[id="inputlastname"]').simulate('change', { target: { value: 'bombay_lastname' } })
        wrapper.find('[id="inputaddress"]').simulate('change', { target: { value: 'bombay_address' } })
        wrapper.instance().handleOk()
        expect(window.alert).toBeCalledWith('please full fill infomation')

        wrapper.find('[id="inputfirstname"]').simulate('change', { target: { value: 'bombay_firstname' } })
        wrapper.find('[id="inputlastname"]').simulate('change', { target: { value: '' } })
        wrapper.find('[id="inputaddress"]').simulate('change', { target: { value: 'bombay_address' } })
        wrapper.instance().handleOk()
        expect(window.alert).toBeCalledWith('please full fill infomation')

        wrapper.find('[id="inputfirstname"]').simulate('change', { target: { value: 'bombay_firstname' } })
        wrapper.find('[id="inputlastname"]').simulate('change', { target: { value: 'bombay_lastname' } })
        wrapper.find('[id="inputaddress"]').simulate('change', { target: { value: '' } })
        wrapper.instance().handleOk()
        expect(window.alert).toBeCalledWith('please full fill infomation')
    })
})