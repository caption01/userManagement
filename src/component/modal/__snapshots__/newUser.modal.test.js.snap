// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NewUserModal expect render newUserModal 1`] = `
<div
  className="edit-modal-container"
>
  <Modal
    confirmLoading={false}
    footer={
      Array [
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
        >
          Return
        </Button>,
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
          type="primary"
        >
          Submit
        </Button>,
      ]
    }
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    title="Add New User"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <p>
      First_Name : 
      <Input
        allowClear={true}
        id="inputfirstname"
        onChange={[Function]}
        placeholder="insert your first name"
        type="text"
        value=""
      />
       
    </p>
    <p>
      Last_Name : 
      <Input
        allowClear={true}
        id="inputlastname"
        onChange={[Function]}
        placeholder="insert your last name"
        type="text"
        value=""
      />
       
    </p>
    <p>
      Address : 
      <Input
        allowClear={true}
        id="inputaddress"
        onChange={[Function]}
        placeholder="insert your address"
        type="text"
        value=""
      />
       
    </p>
  </Modal>
</div>
`;
